src/libYARP_dev/include/yarp/dev/GenericVocabs.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * Copyright (C) 2006-2010 RobotCub Consortium
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_DEV_GENERIC_VOCABS_H
+#define YARP_DEV_GENERIC_VOCABS_H
+
+#include <yarp/os/Vocab.h>
+// Common
+constexpr yarp::conf::vocab32_t VOCAB_SET               = yarp::os::createVocab('s','e','t');
+constexpr yarp::conf::vocab32_t VOCAB_GET               = yarp::os::createVocab('g','e','t');
+constexpr yarp::conf::vocab32_t VOCAB_IS                = yarp::os::createVocab('i','s');
+constexpr yarp::conf::vocab32_t VOCAB_OK                = yarp::os::createVocab('o','k');
+constexpr yarp::conf::vocab32_t VOCAB_FAILED            = yarp::os::createVocab('f','a','i','l');
+constexpr yarp::conf::vocab32_t VOCAB_ERR               = yarp::os::createVocab('e','r','r');
+constexpr yarp::conf::vocab32_t VOCAB_HELP              = yarp::os::createVocab('h','e','l','p');
+constexpr yarp::conf::vocab32_t VOCAB_QUIT              = yarp::os::createVocab('q','u','i','t');
+constexpr yarp::conf::vocab32_t VOCAB_NOT               = yarp::os::createVocab('n','o','t');
+constexpr yarp::conf::vocab32_t VOCAB_REMOVE            = yarp::os::createVocab('r','m');
```

Comment:
Are we sure that this definition are not violating the One Definition Rule if they are included in several compilation unit? Should be declared `inline`? 

References: 
* https://en.cppreference.com/w/cpp/language/definition 
* https://www.codeproject.com/Articles/762413/The-One-Definition-Rule-in-Cplusplus-and-Cplusplus

Reply:
I'm not sure if this applies to `constexpr`.
https://timsong-cpp.github.io/cppwp/dcl.constexpr
> A function or static data member declared with the constexpr specifier is implicitly an inline function or variable (7.1.6).

This is not a "static data member", therefore I have no idea, but I think it's easy to test using `nm` if these symbols are exported in the `.o` objects (@Nicogene)

Also inline variables are c++17, therefore this is not the right way to handle it in c++11.

The biggest problem that I see (and that is one of the reasons why I was against the `GenericVocabs.h` file, is that all these expressions are re-evaluated every time the file is included...

Reply:
I checked in the `.o` there is no trace of Vocabs, probably `constexpr` do the magic.

Reply:
I am not sure I fully understood the reply, even because I am not sure of the relation between symbol visibility and ODR. However if you (and @Nicogene) checked this, I am happy to trust you.

Reply:
Ok, let's go, apparently this is the same solution suggested by the new Visual Studio tool: https://blogs.msdn.microsoft.com/vcblog/2018/06/26/convert-macros-to-constexpr/ .